CIP-0146/README.md

```diff
@@ -0,0 +1,220 @@
+- Additional optional metadata can be added to the `1854` metadatum label to describe the multisig.
+- Optional icon fields is a URL to a non-animated image file, maximum 40kb in size.
+
+### Data Types
+
+#### MultiSigRegistration
+Label `1854` metadata.
+```ts
+interface MultiSigRegistration {
+  types: ScriptType[];
+  name?: string;
+  description?: string;
+  icon?: string;
+  participants?: MultiSigParticipants;
+}
+```
+
+#### ScriptType
+A number representing a specific type (role). This corresponds to the key derivation path role (`x`).
+```ts
+type ScriptType = number;
+```
+
+#### MultiSigParticipants
+A mapping between key credential and participant details.
+```ts
+interface MultiSigParticipants {
+  [key: string]: MultiSigParticipant;
+}
+```
+
+#### MultiSigParticipant
+Multisig participant details.
+```ts
+interface MultiSigParticipant {
+  name: string;
+  description?: string;
+  icon?: string;
+}
+```
+
+#### OffChainMultiSigRegistration
+The hex-encoded bytes for a transaction auxiliary data `metadata` and `native_script` array.
+```ts
+interface OffChainMultiSigRegistration {
+  metadata: string;
+  native_scripts: string;
+}
+```
+
+### Registration
+
+After the multisig wallet has been defined according to the Cardano native script standard, and adhering to the
+[rules](#specification), either a registration transaction can be put on the blockchain or a JSON download provided for 
+off-chain sharing. 
+
+>Note that the registration **must** use previously unused keys in the scripts **if** registered in a transaction. 
+
+The transaction auxiliary data metadata ([MultiSigRegistration](#multisigregistration)) should be formatted using 
+NoConversions JSON schema.
+
+If using the CSL library, this can be accomplished with helper functions `encode_json_str_to_metadatum` and 
+`decode_metadatum_to_json_str` specifying `MetadataJsonSchema.NoConversions` as the JSON schema.
+
+Input Output Global (IOG) documentation also describe the
+[no schema](https://github.com/input-output-hk/cardano-node-wiki/blob/main/docs/reference/tx-metadata.md#no-schema)
+JSON mapping in more detail.
+
+#### Transaction
+
+A registration transaction is the most user-friendly alternative as it allows for automatic multisig wallet discovery 
+by its participants. 
+
+#### Off-chain download
+
+The off-chain JSON file should have the format of `OffChainMultiSigRegistration`.
+
+### Discovery
+
+Discovering registered multisig wallets on the blockchain that has public keys included for a participant wallet 
+can be done in the following way.
+
+- Derive Ed25519 verification keys from path `1854'/1815'/0'/x/y`.
+- Create key credentials, `blake2b-224` hash digests of derived keys in previous step.
+- Search for multisig registration transactions on the blockchain that contain metadata with metadatum label `1854` 
+  and key credentials matching participant wallet. Only the first (oldest) encountered match should be returned.

```

Comment:
Is there the possibility for a replay attack here, where someone submits many of the same registration that they have observed from on-chain?

Reply:
This is why the key rotation was added to the specification to prevent just that. So only the first registration where a key is used is valid. On creation of a multisig wallet you must first check for next unused key for each participant .